charts/dscp-ipfs-v2/templates/statefulset.yaml

```diff
@@ -0,0 +1,320 @@
+              subPath: {{ .Values.persistence.subPath }}
+              {{- end }}
+          env:
+            - name: IPFS_PATH
+              value: {{ .Values.persistence.mountPath }}
+            {{- if .Values.bootNodeAddress }}
+            - name: BOOT_NODE_ADDRESS
+              value: {{ .Values.ipfs.bootNodeAddress }}
+            {{- end}}
+        {{- end }}
+        {{- if and .Values.ipfs.initKeys.enabled (or .Values.ipfs.existingSecret (and .Values.ipfs.publickey .Values.ipfs.privatekey)) }}
+        - name: init-ipfs-keys
+          image: jetbrainsinfra/jq:latest
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.containerSecurityContext.enabled }}
+          securityContext: {{- omit .Values.containerSecurityContext "enabled" | toYaml | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.command "context" $) | nindent 12 }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
+          {{- else }}
+          {{ if .Values.ipfs.keyCommand }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.ipfs.keyCommand "context" $) | nindent 12 }}
+          {{- else }}
+          command:
+            - /bin/sh
+            - -c
+            - -e
+          {{- end }}
+          {{ if .Values.ipfs.keyArgs }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.ipfs.keyArgs "context" $) | nindent 12 }}
+          {{- else }}
+          args:
+            - |
+              echo "$(jq --arg publickey $IPFS_PUBLIC_KEY '.Identity.PeerID = $publickey' {{ .Values.persistence.mountPath }}/config)" > {{ .Values.persistence.mountPath }}/config ;
+              echo "$(jq --arg privatekey $IPFS_PRIVATE_KEY '.Identity.PrivKey = $privatekey' {{ .Values.persistence.mountPath }}/config)" > {{ .Values.persistence.mountPath }}/config ;
+          {{- end }}
+          {{- end }}
+          volumeMounts:
+            - name: {{ default (include "common.names.fullname" .) .Values.persistence.existingClaim }}
+              mountPath: {{ .Values.persistence.mountPath }}
+              {{- if .Values.persistence.subPath }}
+              subPath: {{ .Values.persistence.subPath }}
+              {{- end }}
+          env:
+            - name: IPFS_PATH
+              value: {{ .Values.persistence.mountPath }}
+            - name: IPFS_PUBLIC_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "dscp-ipfs.secretName" . }}
+                  key: {{ include "dscp-ipfs.publicKeyKey" . }}
+            - name: IPFS_PRIVATE_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "dscp-ipfs.secretName" . }}
+                  key: {{ include "dscp-ipfs.privateKeyKey" . }}
+        {{- end }}
```

Comment:
This has been made conditional, the advantage is that we only have to provide a known public and private key for the Bootnode, other other ipfs nodes can use the keys generated in the `init-ipfs-config` for their public and private keypair.